.goreleaser.yml

```diff
@@ -106,6 +106,23 @@ signs:
   - artifacts: checksum
     args: ["--batch", "-u", "{{ .Env.GPG_FINGERPRINT }}", "--output", "${signature}", "--detach-sign", "${artifact}"]
 
+brews:
+  - tap:
+      owner: ossf
+      name: homebrew-tap
```

Comment:
How will this action have permission to write to another repository?

Reply:
This [token](https://github.com/ossf/scorecard/blob/main/.github/workflows/goreleaser.yaml#L56) should have enough permissions to do that, right, if not, it'll fail.

Reply:
OK. Thanks

Reply:
@naveensrinivasan I think we should create PAT to write another repository, the default actions token only has perms in the current repo 🙋🏻‍♂️ I asked to the maintainer of the GoReleaser project, he told me like that

Reply:
We need to create one and store it on GitHub secret, maybe name it like 'GH_PAT', then we can use it 🙋🏻‍♂️

Reply:
Can you do that @naveensrinivasan ?

Reply:
or we can create a PAT specific to the repository `homebrew-tap`, and set it to the `token` variable of the `tap` section in `.goreleaser.yml` file.

```shell
 tap:
      # Optionally a token can be provided, if it differs from the token provided to GoReleaser
      token: "{{ .Env.HOMEBREW_TAP_GITHUB_TOKEN }}"
```

Reply:
Is this how folks typically set this up? Seems complicated and also somewhat a security issue since we need a write access token.

Reply:
You won't be right token here, you will store it in GitHub Secret, then pass it to the GoReleaser as an environment variable as you did, I just put it as an example to show you How we can use it, the only thing that we need to do is creating a PAT and store it in GitHub Secret, and name it GH_PAT, that's all I need right now 🙋🏻‍♂️

Reply:
I am not comfortable having a token that can write another repository TBH. That is my take. I will let others chime in.

Reply:
where does the workflow write? I did not know some tokens could write to a repo that is not our own.

Reply:
Let me summarize the steps we should take to accomplish this:

1. We are now creating a HomeBrew formula for the scorecard, so we need to upload it to its repository, which we named `homebrew-tap.`
2. To do that, we have to create a PAT that has enough permissions to push this repository. We can create PAT just for the `homebrew-tap` repository and store it in `GitHub Secrets` of the `scorecard` repository, and maybe we can name it as `GH_PAT`.
3. Then, I'll update the `token` variable of the `tap` section with it.
4. That's it. We don't worry about anything, right? Because everything will be stored in GitHub Secret as we did.

So, if you agree to this, please feel free to create a PAT for the `homebrew-tap` repository to be able to push there, and save it in `GitHub Secrets` of the `scorecard` repository, and name it as `GH_PAT`. 🙋🏻‍♂️ @naveensrinivasan @laurentsimon

Reply:
please see 0ba7c02.

Reply:
Also, we created a dedicated section for this problem on the GoReleaser website. 👇 

https://goreleaser.com/errors/resource-not-accessible-by-integration/#resource-not-accessible-by-integration

Reply:
Wondering if there's an alternative to PAT here? Something which only allows write access to `homebrew-tap` repo. PAT has the disadvantage that write access is provided to all repos owned by the user. If hypothetically this token gets leaked, there is a lot at stake. But at least with a token which only has write access to `homebrew-tap` repo, the blast radius is reduced.

Also in terms of maintainability and developer experience, it would be great not to have a single developer's account as the point of failure. If its a key which can be owned/maintained by multiple developers I would feel more at ease.

I guess this is a feature request for the `goreleaser` team. And I personally think it's very much required. Asking developers to share their write-access PATs in GitHub workflows is not a good practice IMO.

Reply:
To summarize, here's my take on this:

1. I would *really really* prefer not to go the PAT route. If this is doable, IMO we should wait until then.
2. If not, I'm ok biting the bullet for now and submitting this PR *as long as* there is some commitment from the `goreleaser` team that there will be a better and more secure alternative in the future.

Reply:
> Something which only allows write access to homebrew-tap repo.
This is exactly what I want right now.

Reply:
I need a PAT specific for the `homebrew-tap` repository, as I said above, please[see](https://github.com/ossf/scorecard/pull/1376#discussion_r764379160).

Reply:
PTAL @caarlos0 🙋🏻‍♂️